neutronclient/client.py

```diff
@@ -239,6 +248,8 @@ def authenticate(self):
             self._authenticate_keystone()
         elif self.auth_strategy == 'noauth':
             self._authenticate_noauth()
+        elif self.auth_plugin:
+            self._plugin_auth(self.auth_url)
```

Comment:
default is keystone.  if a plugin is installed, it should overwrite the default IMO.  that means plugin should have precedence.